docs/server/source/production-deployment-template-tendermint/architecture.rst

```diff
@@ -0,0 +1,204 @@
+  |                                    |                                                |                                    |
+  |                                    |                                                |                                    |
+  |      3Scale                        |                                                |    MongoDB Cloud                   |
+  |                                    |                                                |                                    |
+  |                                    |                                                |                                    |
+  |                                    |                                                |                                    |
+  +------------------------------------+                                                +------------------------------------+
+
+
+
+.. note::
+  The arrows in the diagram represent the client-server communication. For
+  example, A-->B implies that A initiates the connection to B.
+  It does not represent the flow of data; the communication channel is always
+  fully duplex.
+
+
+NGINX: Entrypoint and Gateway
+-----------------------------
+
+We use an NGINX as HTTP proxy on port 443 (configurable) at the cloud
+entrypoint for:
+
+#. Rate Limiting: We configure NGINX to allow only a certain number of requests
+   (configurable) which prevents DoS attacks.
+
+#. HTTPS Termination: The HTTPS connection does not carry through all the way
+   to BigchainDB and terminates at NGINX for now.
+
+#. Request Routing: For HTTPS connections on port 443 (or the configured BigchainDB public api port),
+   the connection is proxied to:
+
+   #. OpenResty Service if it is a POST request.
+   #. BigchainDB Service if it is a GET request.
+
+
+We use an NGINX TCP proxy on port 27017 (configurable) at the cloud
+entrypoint for:
+
+#. Rate Limiting: We configure NGINX to allow only a certain number of requests
+   (configurable) which prevents DoS attacks.
+
+#. Request Routing: For connections on port 27017 (or the configured MongoDB
+   public api port), the connection is proxied to the MongoDB Service.
+
+
+OpenResty: API Management, Authentication and Authorization
+-----------------------------------------------------------
+
+We use `OpenResty <https://openresty.org/>`_ to perform authorization checks
+with 3scale using the ``app_id`` and ``app_key`` headers in the HTTP request.
+
+OpenResty is NGINX plus a bunch of other
+`components <https://openresty.org/en/components.html>`_. We primarily depend
+on the LuaJIT compiler to execute the functions to authenticate the ``app_id``
+and ``app_key`` with the 3scale backend.
+
+
+MongoDB: Standalone
+-------------------
+
+We use MongoDB as the backend database for BigchainDB.
+In a multi-node deployment, MongoDB members communicate with each other via the
```

Comment:
When you write "multi-node" here, what do you mean? Several BigchainDB nodes, or several MongoDB nodes inside a single BigchainDB node (i.e. a local MongoDB cluster)?

In any case, why would a local MongoDB be exposed on a public port?

Reply:
I think this needs to be removed, there is no replica set, hence no MongoDB members and no communication.

Reply:
:+1: